assets/app/scripts/directives/fromFile.js

```diff
@@ -0,0 +1,407 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .directive("fromFile", function($q,
+                                  $uibModal,
+                                  $location,
+                                  $filter,
+                                  CachedTemplateService,
+                                  AlertMessageService,
+                                  Navigate,
+                                  TaskList,
+                                  DataService,
+                                  APIService) {
+    return {
+      restrict: "E",
+      scope: false,
+      templateUrl: "views/directives/from-file.html",
+      controller: function($scope) {
+        var aceEditorSession;
+        var humanizeKind = $filter('humanizeKind');
+        $scope.aceError = false;
+        TaskList.clear();
+
+        $scope.aceLoaded = function(editor) {
+          aceEditorSession = editor.getSession();
+          aceEditorSession.setOption('tabSize', 2);
+          aceEditorSession.setOption('useSoftTabs', true);
+          editor.setDragDelay = 0;
+          editor.$blockScrolling = Infinity;
+        };
+
+        var checkAceForErrors = function() {
+          var editorAnnotations = aceEditorSession.getAnnotations();
+          var errors = [];
+          _.forEach(editorAnnotations, function(annot) {
+            if (annot.type === 'error') {
+              errors.push(annot);
+            }
+          });
+          if (_.isEmpty(errors)) {
+            $scope.aceError = false;
+          } else {
+            $scope.aceError = true;
+          }
```

Comment:
I might change the scope var name to something more descriptive like `editorErrorAnnotation`.